packages/babel-generator/src/generators/jsx.js

```diff
@@ -1,8 +1,9 @@
 export function JSXAttribute(node: Object) {
   this.print(node.name, node);
   if (node.value) {
+    const raw = node.extra && node.extra.raw;
     this.token("=");
-    this.print(node.value, node);
+    this.print(raw || node.value, node);
```

Comment:
I know this PR has a few comments against the current implementation already, but I also want to add that this could in some cases lead to a breaking change. Say you had a transform that worked on these nodes that changed the `.value`. With this PR, the mutated value would be ignored in the final output since `raw` would exist. You'd want to use `getPossibleRaw`: https://github.com/babel/babel/blob/672adba9a14aa3b4d4676ef6b8699616d2062a55/packages/babel-generator/src/printer.js#L390 so that `value` is compared to `rawValue`